src/renderer/coremods/utilityClasses/index.ts

```diff
@@ -0,0 +1,138 @@
+    "ClientThemesBackgroundStore",
+  );
+  if (!ClientThemesBackgroundStore) {
+    throw new Error("Failed to find ClientThemesBackgroundStore!");
+  }
+  const ThemeIDMap = getByProps<ThemeIDMap>("MINT_APPLE");
+  if (!ThemeIDMap) {
+    throw new Error("Failed to find ThemeIDs module!");
+  }
+
+  // update theme attribute when theme changes
+  ClientThemesBackgroundStore.addChangeListener(() => {
+    onNitroThemeChange(ClientThemesBackgroundStore, ThemeIDMap);
+  });
+  onNitroThemeChange(ClientThemesBackgroundStore, ThemeIDMap);
+}
+
+interface MessageComponent {
+  props: { id: string | undefined }; // may not be there for non-message components
+  type: {
+    type: (msg: { message: Message }) => React.ReactElement;
+  };
+}
+
+async function messageDataAttributes(): Promise<void> {
+  const MessagesComponent = await waitForModule<{
+    type: () => React.ReactElement;
+  }>(filters.bySource(".content.id)"));
+
+  // the Message component isn't exported, so it must be extracted like this
+  const uninjectMessagesComponent = inject.after(MessagesComponent, "type", (_, res) => {
+    uninjectMessagesComponent();
+
+    const MessageListComponent = res.props.children.type;
+    const uninjectMessageList = inject.after(MessageListComponent, "type", (_, res) => {
+      const messagesArray = res.props.children.props.children[1].props.children[1].props
+        .children[1] as MessageComponent[];
+      const messageElement = messagesArray.find((e) => e.props.id !== undefined);
+
+      // messageElement isn't found when first loading the message list.
+      if (messageElement) {
+        uninjectMessageList();
+        // found a message component, inject into it
+        inject.after(messageElement.type, "type", ([{ message }], res) => {
+          const { props } = res.props.children.props.children;
+          props["data-is-author-self"] = message.author.id === users.getCurrentUser().id;
+          props["data-is-author-bot"] = message.author.bot;
+          // webhooks are also considered bots
+          if (message.author.bot) {
+            props["data-is-author-webhook"] = Boolean(message.webhookId);
+          }
+          props["data-author-id"] = message.author.id;
+          props["data-message-type"] = message.type; // raw enum value, seems consistent
+          if (message.blocked) props["data-is-blocked"] = "true";
+          return res;
+        });
+      }
+
+      return res;
+    });
+    return res;
+  });
+}
+
+function addHtmlClasses(): void {
+  if (!html.classList.contains("replugged")) {
+    html.classList.add("replugged");
+  }
+}
+
+export async function start(): Promise<void> {
+  tabBarItemId();
+  nitroThemeClass();
+  await messageDataAttributes();
+
+  // generic stuff
+  const observer = new MutationObserver(addHtmlClasses);
+  observer.observe(html, { attributeFilter: ["class"] });
+}
+
+export function stop(): void {
+  inject.uninjectAll();
```

Comment:
Remove the observer and at least some of the classes

Reply:
removed the observer, the `replugged` class, and the `data-nitro-theme` attribute. the rest of the changes get removed when the (no longer) injected elements are re-rendered.